gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -851,6 +852,20 @@ private final PreparedStatement prepMinion(
                                             }
                                           }
 
+                                          if (routeQuery && qinfo != null && qinfo.isDML() && qinfo.isInsert() && ((InsertQueryInfo)qinfo).isPutDML()) {
+                                            InsertNode in = qt instanceof InsertNode ? ((InsertNode)qt) : null;
+                                            TableDescriptor ttd = in != null ? in.targetTableDescriptor : null;
+                                            String table = ttd.getQualifiedName();
+                                            Region region = Misc.getRegionForTable(table.replaceAll("\"", ""), true);
+                                            GemFireContainer container = (GemFireContainer)region.getUserAttribute();
+                                            boolean isColumnTable = container.isRowBuffer();
+                                            if (isColumnTable) {
+                                              return getPreparedStatementForSnappy(true, statementContext, lcc,
+                                                  false, checkCancellation,
+                                                  DML_TABLE_PATTERN.matcher(source).find(), null);
```

Comment:
Instead of pattern match (DML_TABLE_PATTERN.matcher(source).find()), just pass true here

Reply:
ok